src/raw/unix.rs

```diff
@@ -414,6 +418,32 @@ impl Socket {
     }
 
 
+    ///Sets whether this socket will be inherited to child porcesses or not.
+    pub fn set_inheritable(&self, value: bool) -> io::Result<()> {
+        // Some (or possibly all?) OS's support the `FIOCLEX` and `FIONCLEX`

```

Comment:
Actually I lied: Apparently `libc` does have definitions for `FIOCLEX` (including macOS!), just not for `FIONCLEX`. That probably means that all we could completely get rid of the `fcntl` approach for unices once the constant values for `FIONCLEX` are figured out.